typing/jkind.mli

```diff
@@ -565,10 +565,20 @@ val set_nullability_upper_bound :
 (** Sets the layout in a jkind. *)
 val set_layout : 'd Types.jkind -> Sort.t Layout.t -> 'd Types.jkind
 
-(** Update a jkind by applying a modality. This modifies the mod-bounds, and propagates
-    the modality to the each of the jkind's with-bounds. *)
+(** Change a jkind to be appropriate for a type that appears under a
+    modality. This means that the jkind will definitely cross the axes
+    modified by the modality, by setting the mod-bounds appropriately
+    and propagating the modality into any with-bounds. *)
 val apply_modality :
-  Mode.Modality.Value.Const.t -> 'd Types.jkind -> 'd Types.jkind
+  Mode.Modality.Value.Const.t -> Types.jkind_l -> Types.jkind_l
+
+(** Change a jkind to be appropriate for an expectation of a type under
+    a modality. This means that the jkind's axes affected by the modality
+    will all be top. The with-bounds are left unchanged. *)
+val apply_modality_to_expected :
+  Mode.Modality.Value.Const.t ->
+  ('l * allowed) Types.jkind ->
+  ('l * allowed) Types.jkind
```

Comment:
idea: maybe `apply_modality_l` and `apply_modality_r`? or `_left` and `_right` if you want to be more verbose

Reply:
(the "expected" suffix threw me for a bit of a loop originally)